pipeline-modules/deployment-service/azure/run-bash-script.yaml

```diff
@@ -0,0 +1,92 @@
+provisioner: azure
+name: run-bash-script
+version: 1
+revision: 1
+displayName: Execute a bash script
+description: Execute a bash script from a file
+target: "deployment-template"
+keywords:
+  - bash
+  - linux
+author: CloudCover
+meta:
+  inputArtifactType:
+    - GitCode
+
+inputs:
+  properties:
+    bash_script:
+      title: Bash script
+      description: This the filename of the bash script to run
+      type: string
+    script_parameters:
+      title: Parameters to pass to the bash script
+      type: array
+      default: []
+      items:
+        type: string
+    repo_dir:
+      title: Repo Dir
+      description: Repo Dir
+      type: string
+    job_type:
+      title: Job Type
+      description: This is to deploy or undeploy application
+      type: string
+      default: deploy
+    pipeline_summary_var:
+      title: Pipeline Summary
+      description: This will create pipeline summary as per user request
+      type: array
+      default: []
+      items:
+        type: object
+        required: [Command, Name]
+        properties:
+          Command:
+            title: Command
+            type: string
+          Name:
+            title: Name
+            type: string
+    pipeline_secret_env:
+      title: Global Secret Environment
+      description: Mapping of globally available secret variable id.
+      type: object
+      default: {}
+  required:
+    - bash_script
+  internal:
+    - pipeline_secret_env
+    - pipeline_summary_var
+    - repo_dir
+    - job_type
+template: |
+  {% set workspace = '$(Build.SourcesDirectory)' %}
+  {% set pipeline_env_file = workspace|add:'/.env' %}
+  {% set pipeline_secret_env_file = workspace|add:'/.secret.env' %}
+  {% set source_code_path = workspace|add:'/source_code' %}
+
+  steps:
+  - script: |
+      export CODEPIPES_DEPLOY_ACTION={{ job_type }} ;
```

Comment:
I believe that here instead of using export you can just add it to the `env:` section for this step below (you could get rid of the `{% if pipeline_secret_env %}` statement as there would always be one variable)

Reply:
I updated all 3 modules with the same pattern and tested - looks good -thx